source/extensions/filters/http/cache/inline_headers_handles.h

```diff
@@ -9,46 +9,93 @@ namespace Extensions {
 namespace HttpFilters {
 namespace Cache {
 
-// Request headers inline handles
 inline Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::RequestHeaders>
-    authorization_handle(Http::CustomHeaders::get().Authorization);
+    authorization_handle() {
+  static Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::RequestHeaders> handle(Http::CustomHeaders::get().Authorization);
```

Comment:
Can we do three changes here:

1. have one lazy-init for all the cache headers, held in a struct.
2. at the call-site you can grab the lazy-init struct once and use that for all the headers.
3. lazy-init using the Envoy pattern for that. See source/common/common/macros.h, CONSTRUCT_ON_FIRST_USE. Note that it lazy-constructs a pointer rather than lazy-constructing a value, which is better because it will be left dangling at the end of the program, rather than having a potential out-of-order destruction issue.

See https://github.com/envoyproxy/envoy/blob/main/STYLE.md for details:
> The Google C++ style guide points out that non-PoD static and global variables are forbidden. This includes types such as std::string. We encourage the use of the advice in the C++ FAQ on the static initialization fiasco for how to best handle this.